src/bp/core/services/dialog/dialog-engine.ts

```diff
@@ -223,6 +230,67 @@ export class DialogEngine {
     }
   }
 
+  private copyVarsToParent(flow: FlowView, currentNode: FlowNode, event: IO.IncomingEvent) {
+    const { workflows } = event.state.session
+    const { createVariable } = event.state
+
+    const childFlowVars = workflows[currentNode.flow!.replace('.flow.json', '')]?.variables
+    const outputVars = flow.variables?.filter(x => x.isOutput)
+
+    if (!outputVars || !childFlowVars) {
+      return
+    }
+
+    outputVars.forEach(v => {
+      const ouputVariableName = currentNode.subflow?.out?.[v.name]
+      if (!ouputVariableName) {
+        return
+      }
+
+      const cv = childFlowVars[v.name] as BoxedVariable<any>
+      if (cv) {
+        createVariable(ouputVariableName, cv.value, cv.type)
+      }
+    })
+  }
+
+  private sendVarsToChild(
+    flow: FlowView,
+    nextFlowName: string,
+    event: IO.IncomingEvent,
+    inputs?: { [variable: string]: SubWorkflowInput }
+  ) {
+    const { workflow, createVariable } = event.state
```

Comment:
You have the complete event here, you can access the inputs directly instead of passing them as an argument